optuna/integration/allennlp.py

```diff
@@ -9,17 +10,156 @@
+    if pruner_class is None:
+        return None
+
+    pruner_params = _get_environment_variables_for_pruner()
+    pruner = getattr(optuna.pruners, pruner_class, None)
+
+    if pruner is None:
+        return None
+
+    return pruner(**pruner_params)
+
+
+def _infer_and_cast(value: Optional[str]) -> Optional[Union[str, int, float, bool]]:
+    """Infer and cast a string to desired types.
+
+    We are only able to set strings as environment variables.
+    However, parameters of a pruner could be integer, float,
+    boolean, or else. We infer and cast environment variables
+    to desired types.
+
+    """
+    if value is None:
+        return None
+
+    try:
+        return int(value)
+    except ValueError:
+        try:
+            return float(value)
+        except ValueError:
+            if value == "True":
+                return True
+            elif value == "False":
+                return False
+
+    return value
+
+
+def _get_environment_variables_for_trial() -> Dict[str, Optional[str]]:
+    return {
+        "study_name": os.getenv(_add_ppid("OPTUNA_ALLENNLP_STUDY_NAME")),
+        "trial_id": os.getenv(_add_ppid("OPTUNA_ALLENNLP_TRIAL_ID")),
+        "storage": os.getenv(_add_ppid("OPTUNA_ALLENNLP_STORAGE_NAME")),
+        "monitor": os.getenv(_add_ppid("OPTUNA_ALLENNLP_MONITOR")),
+    }
+
+
+def _get_environment_variables_for_pruner() -> Dict[str, Optional[Union[str, int, float, bool]]]:
+    keys = os.getenv(_add_ppid("OPTUNA_ALLENNLP_PRUNER_KEYS"))
+    if keys is None:
+        return {}
+
+    kwargs = {}
+    for key in keys.split(","):
+        parameter_key = "OPTUNA_ALLENNLP_{}".format(key)
+        value = os.getenv(_add_ppid(parameter_key))
+        kwargs[key] = _infer_and_cast(value)
+
+    return kwargs
+
+
+def _fetch_pruner_config(trial: optuna.Trial) -> Dict[str, Any]:
+    pruner = trial.study.pruner
+    kwargs = {}  # type: Dict[str, Any]
+
+    if isinstance(pruner, optuna.pruners.HyperbandPruner):
+        kwargs["min_resource"] = pruner._min_resource
+        kwargs["max_resource"] = pruner._max_resource
+        kwargs["reduction_factor"] = pruner._reduction_factor
+    elif isinstance(pruner, optuna.pruners.PercentilePruner):
+        kwargs["percentile"] = pruner._percentile
+        kwargs["n_startup_tirals"] = pruner._n_startup_trials
+        kwargs["n_warmup_steps"] = pruner._n_warmup_steps
+        kwargs["interval_steps"] = pruner._interval_steps
+    elif isinstance(pruner, optuna.pruners.ThresholdPruner):
+        kwargs["lower"] = pruner._lower
+        kwargs["upper"] = pruner._upper
+        kwargs["n_warmup_steps"] = pruner._n_warmup_steps
+        kwargs["interval_steps"] = pruner._interval_steps
+    elif isinstance(pruner, optuna.pruners.SuccessiveHalvingPruner):
+        kwargs["min_resource"] = pruner._min_resource
+        kwargs["reduction_factor"] = pruner._reduction_factor
+        kwargs["min_early_stopping_rate"] = pruner._min_early_stopping_rate
+    else:
```

Comment:
`NopPruner` is not supported. Could you add it?
```suggestion
    elif isinstance(pruner, optuna.pruners.NopPruner):
        pass
    else:
```